api/src/main/java/net/wesjd/anvilgui/version/VersionProvider.java

```diff
@@ -0,0 +1,85 @@
+/*
+MIT License
+
+Copyright (c) 2018-2020 PaperMC
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+ */
+package net.wesjd.anvilgui.version;
+
+
+import java.util.regex.MatchResult;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.bukkit.Bukkit;
+
+/**
+ * Partially copied from <a href="https://github.com/PaperMC/PaperLib/blob/master/src/main/java/io/papermc/lib/environments/Environment.java#L56">here</a>.
+ */
+final class VersionProvider {
+    private static final Pattern VERSION_PATTERN =
+            Pattern.compile("(?i)\\(MC: (\\d)\\.(\\d+)\\.?(\\d+?)?(?: (Pre-Release|Release Candidate) )?(\\d)?\\)");
+
+    private final int major;
+    private final int minor;
+    private final int patch;
+
+    VersionProvider() {
+        this(Bukkit.getVersion());
+    }
+
+    VersionProvider(final String bukkitVersion) {
+        final Matcher matcher = VersionProvider.VERSION_PATTERN.matcher(bukkitVersion);
+        int major = 1;
+        int minor = 0;
+        int patch = 0;
+        if (matcher.find()) {
+            final MatchResult matchResult = matcher.toMatchResult();
+            try {
+                major = Integer.parseInt(matchResult.group(1), 10);
+            } catch (final Exception ignored) {
```

Comment:
Why are these exceptions ignored? Isn't the version unable to be parsed correctly? Shouldn't a new error like `CannotParseMinecraftVersionException` or something be thrown?